### ISSUE TITLE
The executable bit on documentation files should not be set

### ISSUE THREAD

**Issue by justindho** [aws/aws-cli#6972]
### Describe the bug

The executable bit on documentation files should not be set.

### Expected Behavior

The executable bit on documentation files should not be set.

### Current Behavior

The executable bit on some documentation files is currently set.

### Reproduction Steps

```$ ls -lah awscli/examples/<service>```

For example:
```
$ ls -lah /awscli/examples/alexaforbusiness
total 40
drwxr-xr-x    7 user  staff   224B May 18 12:43 .
drwxr-xr-x  172 user  staff   5.4K May 18 14:25 ..
-rwxr-xr-x    1 user  staff   702B May 18 12:43 create-network-profile.rst
-rwxr-xr-x    1 user  staff   553B May 18 12:43 delete-network-profile.rst
-rwxr-xr-x    1 user  staff   871B May 18 12:43 get-network-profile.rst
-rwxr-xr-x    1 user  staff   1.5K May 18 12:43 search-network-profiles.rst
-rwxr-xr-x    1 user  staff   597B May 18 12:43 update-network-profile.rst
```

shows that the executable bit is set on the .rst files.

### Possible Solution

#5490

### Additional Information/Context

A test should be added to make sure that no documentation files have the executable bit set.

### CLI version used

any

### Environment details (OS name and version, etc.)

any

**Comment by chirayuk** [comment]
TLDR: The executable bit should not be set on library files either (in the linux distributions.). Probably worth fixing at the same time as this issue.

---

The libraries (e.g. `aws/dist/libz.so.1`) in the distributed zip files (e.g. `https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip`, linux) should not have their executable bits set.  Could be fixed opportunistically—it doesn't cause breakages.

Here's what they look like.

```
root@dev01 ~ # mkdir ~/test && cd ~/test
root@dev01 ~/test # curl -LO "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 63.5M  100 63.5M    0     0   235M      0 --:--:-- --:--:-- --:--:--  236M

root@dev01 ~/test # unzip awscli-exe-linux-x86_64.zip 'aws/dist/lib*.so.*'
Archive:  awscli-exe-linux-x86_64.zip
  inflating: aws/dist/libpython3.13.so.1.0
  inflating: aws/dist/libz.so.1
  inflating: aws/dist/liblzma.so.5
  inflating: aws/dist/libbz2.so.1
  inflating: aws/dist/libffi.so.6
  inflating: aws/dist/libuuid.so.1
  inflating: aws/dist/libreadline.so.6
  inflating: aws/dist/libtinfo.so.5
  inflating: aws/dist/libsqlite3.so.0

root@dev01 ~/test # ls -l aws/dist/
total 41668
-rwxr-xr-x 1 root root    68192 Jul 31 18:10 libbz2.so.1
-rwxr-xr-x 1 root root    32328 Jul 31 18:10 libffi.so.6
-rwxr-xr-x 1 root root   157440 Jul 31 18:10 liblzma.so.5
-rwxr-xr-x 1 root root 40382392 Jul 31 18:10 libpython3.13.so.1.0
-rwxr-xr-x 1 root root   285136 Jul 31 18:10 libreadline.so.6
-rwxr-xr-x 1 root root  1443704 Jul 31 18:10 libsqlite3.so.0
-rwxr-xr-x 1 root root   174576 Jul 31 18:10 libtinfo.so.5
-rwxr-xr-x 1 root root    20064 Jul 31 18:10 libuuid.so.1
-rwxr-xr-x 1 root root    90160 Jul 31 18:10 libz.so.1
root@dev01 ~/test #
```

https://github.com/jdx/mise/discussions/5858
